src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php

```diff
@@ -50,7 +52,7 @@ public function buildView(FormView $view, FormInterface $form, array $options)
         $view->vars['money_pattern'] = self::getPattern($options['currency']);
 
         if ($options['html5']) {
-            $view->vars['type'] = 'number';
+            $view->vars['type'] = $options['format'] === 'integer' ? 'integer' : 'number';
```

Comment:
If the `divisor` option has a value greater than 1, we still need a `number` input here, right?